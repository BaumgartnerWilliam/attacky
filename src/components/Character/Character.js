import React from 'react';
import { Col, Row, Container } from 'react-bootstrap';

import Dice from '../Dice/Dice';
import HealthBar from '../HealthBar/HealthBar';
import AttackIndicator from '../AttackIndicator/AttackIndicator';
import CharacterModel from './CharacterModel/CharacterModel';

const Character = ({ hp, dice, isRollingDice, lowHp, isPlayer, children, isAttacking }) => (
  <Container data-testid="character" className="character w-50">
    <Row className={`${!isPlayer && 'flex-row-reverse'}`}>
      <Col>
        {isAttacking && <AttackIndicator isPlayer={isPlayer} />}
        <HealthBar value={hp} isDanger={lowHp} />
        {children}
      </Col>
      <Col data-testid="dice">
        <Dice value={dice?.die1} animate={isRollingDice} />
        <Dice value={dice?.die2} animate={isRollingDice} />
      </Col>
    </Row>
  </Container>
);

Character.Model = CharacterModel;

export default Character;
