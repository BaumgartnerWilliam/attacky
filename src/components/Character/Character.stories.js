import React from 'react';
import { Container, Row } from 'react-bootstrap';
import { withKnobs, number, boolean } from '@storybook/addon-knobs';

import idleAsset from '../../assets/monster_idle.png';
import Character from './Character';

export default {
  title: 'Character',
  decorators: [withKnobs]
};

export const withHealthPoints = () => (
  <Container>
    <Character
      hp={number('hp', 70)}
      die1={number('die1', 1)}
      die2={number('die2', 2)}
      lowHp={boolean('lowHp', false)}
      alive={boolean('alive', false)}
      isAttacking={boolean('isAttacking', false)}
      isPlayer={boolean('isPlayer', false)}>
      <Character.Model src={idleAsset} />
    </Character>
  </Container>
);

export const withEnemy = () => (
  <Container>
    <Row>
      <Character
        hp={number('hp', 70)}
        die1={number('die1', 1)}
        die2={number('die2', 2)}
        lowHp={boolean('lowHp', false)}
        alive={boolean('alive', false)}
        isAttacking={boolean('isAttacking', false)}
        isPlayer={true}>
        <Character.Model src={idleAsset} />
      </Character>
      <Character
        hp={number('hp', 70)}
        die1={number('die1', 1)}
        die2={number('die2', 2)}
        lowHp={boolean('lowHp', false)}
        alive={boolean('alive', false)}
        isAttacking={boolean('isAttacking', false)}
        isPlayer={false}>
        <Character.Model src={idleAsset} />
      </Character>
    </Row>
  </Container>
);
