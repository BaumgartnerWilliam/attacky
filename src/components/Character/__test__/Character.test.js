import React from 'react';
import { render } from '@testing-library/react';
import Character from '../Character';

describe('character component', () => {
  it('renders without crashing', () => {
    const { container, getByTestId } = render(<Character />);
    expect(container).toBeInTheDocument();
    expect(getByTestId('character')).toBeInTheDocument();
  });

  it('has a health bar', () => {
    const { getByTestId } = render(<Character />);
    const el = getByTestId('health-bar');
    expect(el).toBeInTheDocument();
  });

  it('has an attacking indicator', () => {
    const { queryByTestId, rerender, getByTestId } = render(<Character />);
    const testid = 'attacking';
    expect(queryByTestId(testid)).toBeNull();
    rerender(<Character isAttacking={true} />);
    expect(getByTestId(testid)).toBeInTheDocument();
  });

  it('renders children', () => {
    const { getByTestId } = render(
      <Character>
        <p data-testid="test"></p>
      </Character>
    );
    const testid = 'test';
    expect(getByTestId(testid)).toBeInTheDocument();
  });

  it('matches snapshot of ennemy attacking', () => {
    const { getByTestId } = render(
      <Character isAttacking={true}>
        <p>children</p>
      </Character>
    );
    expect(getByTestId('character')).toMatchSnapshot();
  });

  it('matches snapshot of player not attacking', () => {
    const { getByTestId } = render(
      <Character isPlayer={true}>
        <p>children</p>
      </Character>
    );
    expect(getByTestId('character')).toMatchSnapshot();
  });
});
