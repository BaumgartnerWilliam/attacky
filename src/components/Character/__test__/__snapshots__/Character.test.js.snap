// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character component matches snapshot of ennemy attacking 1`] = `
<div
  class="character w-50 container"
  data-testid="character"
>
  <div
    class="flex-row-reverse row"
  >
    <div
      class="col"
    >
      <div
        class="justify-content-center row"
        data-testid="attacking"
      >
        <div
          class="spinner-grow spinner-grow-sm text-danger"
        />
      </div>
      <div
        class="justify-content-center row"
      >
        <div
          class="w-50 progress"
          data-testid="health-bar"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            class="progress-bar bg-success progress-bar-animated progress-bar-striped"
            role="progressbar"
          />
        </div>
      </div>
      <p>
        children
      </p>
    </div>
    <div
      class="col"
      data-testid="dice"
    >
      <div
        class="justify-content-center pt-4 row"
        data-testid="dice"
      >
        <img
          class="undefined"
          data-testid="value"
          style="width: 70px;"
        />
      </div>
      <div
        class="justify-content-center pt-4 row"
        data-testid="dice"
      >
        <img
          class="undefined"
          data-testid="value"
          style="width: 70px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`character component matches snapshot of player not attacking 1`] = `
<div
  class="character w-50 container"
  data-testid="character"
>
  <div
    class="false row"
  >
    <div
      class="col"
    >
      <div
        class="justify-content-center row"
      >
        <div
          class="w-50 progress"
          data-testid="health-bar"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            class="progress-bar bg-success progress-bar-animated progress-bar-striped"
            role="progressbar"
          />
        </div>
      </div>
      <p>
        children
      </p>
    </div>
    <div
      class="col"
      data-testid="dice"
    >
      <div
        class="justify-content-center pt-4 row"
        data-testid="dice"
      >
        <img
          class="undefined"
          data-testid="value"
          style="width: 70px;"
        />
      </div>
      <div
        class="justify-content-center pt-4 row"
        data-testid="dice"
      >
        <img
          class="undefined"
          data-testid="value"
          style="width: 70px;"
        />
      </div>
    </div>
  </div>
</div>
`;
